plugin/src/main/java/org/opensearch/sql/plugin/SQLPlugin.java

```diff
@@ -110,12 +151,50 @@ public Collection<Object> createComponents(Client client, ClusterService cluster
     LocalClusterState.state().setClusterService(clusterService);
     LocalClusterState.state().setPluginSettings((OpenSearchSettings) pluginSettings);
 
+    TransportPPLService.getInstance().setPplService(createPPLService((NodeClient) client));
+    TransportSQLService.getInstance().setSqlService(createSQLService((NodeClient) client));
+
     return super
         .createComponents(client, clusterService, threadPool, resourceWatcherService, scriptService,
             contentRegistry, environment, nodeEnvironment, namedWriteableRegistry,
             indexNameResolver, repositoriesServiceSupplier);
   }
 
+  private <T> T doPrivileged(PrivilegedExceptionAction<T> action) {
+    try {
+      return SecurityAccess.doPrivileged(action);
+    } catch (IOException e) {
+      throw new IllegalStateException("Failed to perform privileged action", e);
+    }
+  }
+
+
+  private PPLService createPPLService(NodeClient client) {
+    return doPrivileged(() -> {
+      AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext();
+      context.registerBean(ClusterService.class, () -> clusterService);
+      context.registerBean(NodeClient.class, () -> client);
+      context.registerBean(org.opensearch.sql.common.setting.Settings.class, () -> pluginSettings);
+      context.register(OpenSearchPluginConfig.class);
+      context.register(PPLServiceConfig.class);
+      context.refresh();
+      return context.getBean(PPLService.class);
+    });
+  }
+
+  private SQLService createSQLService(NodeClient client) {
```

Comment:
new SQLService instance should be in request level, not Plugin level.

Reply:
removed